internal/utils_test.go

```diff
@@ -0,0 +1,83 @@
+package internal
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetExecutableFromPluginJSON(t *testing.T) {
+	type args struct {
+		pluginDir string
+	}
+	tcs := []struct {
+		name          string
+		args          args
+		executable    string
+		pluginJSONDir string
+		expected      string
+		err           bool
+	}{
+		{
+			name: "Can retrieve executable from a plugin.json found in provided directory",
+			args: args{
+				pluginDir: "foobar-datasource",
+			},
+			pluginJSONDir: "foobar-datasource",
+			executable:    "gpx_foo",
+			expected:      "gpx_foo",
+		},
+		{
+			name: "Can retrieve executable from plugin.json found in nested 'datasource' directory (when not found in root of provided directory)",
+			args: args{
+				pluginDir: "foobar-app",
+			},
+			pluginJSONDir: filepath.Join("foobar-app", "datasource"),
+			executable:    "gpx_foo",
+			expected:      filepath.Join("datasource", "gpx_foo"),
+		},
+		{
+			name: "Cannot retrieve executable when no plugin.json in root or nested 'datasource' directory",
+			args: args{
+				pluginDir: "foobar-app",
+			},
+			pluginJSONDir: filepath.Join("foobar-app", "foobar-datasource"),
+			err:           true,
+		},
+		{
+			name: "Can support relative paths in executable field of plugin.json (for backwards compatibility)",
+			args: args{
+				pluginDir: "foobar-app",
+			},
+			pluginJSONDir: filepath.Join("foobar-app", "datasource"),
+			executable:    "../gpx_foo",
+			expected:      "gpx_foo",
```

Comment:
I am not sure if this covers the current case for backwards compatibility? Shouldn't this be `expected: "../gpx_foo"`? Maybe I am missing something

Reply:
The old code basically just returned `exe[3:]` therefore stripping the `../` 😆

Reply:
right, right, took me a while to read this one. It's returning `datasource/../gpx_foo` -> `gpx_foo`. 🙃

Reply:
Yeah but that's actually a problem I realized 😅 Here should be actually making sure we ignore path information if it's provided so that we do end up with `datasource/gpx_foo`. Otherwise people can do `../../../gpx_foo`.